tower/src/buffer/worker.rs

```diff
@@ -22,36 +20,38 @@ use tower_service::Service;
 /// implement (only call).
 #[pin_project]
 #[derive(Debug)]
-pub struct Worker<T, Request>
+pub struct Worker<S, Request, E2>
 where
-    T: Service<Request>,
-    T::Error: Into<crate::BoxError>,
+    S: Service<Request>,
+    S::Error: Into<E2>,
 {
-    current_message: Option<Message<Request, T::Future>>,
-    rx: mpsc::Receiver<Message<Request, T::Future>>,
-    service: T,
+    current_message: Option<Message<Request, S::Future, S::Error>>,
+    rx: mpsc::Receiver<Message<Request, S::Future, S::Error>>,
+    service: S,
     finish: bool,
-    failed: Option<ServiceError>,
-    handle: Handle,
+    failed: Option<S::Error>,
+    handle: Handle<S::Error, E2>,
 }
 
 /// Get the error out
 #[derive(Debug)]
-pub(crate) struct Handle {
-    inner: Arc<Mutex<Option<ServiceError>>>,
+pub(crate) struct Handle<E, E2> {
+    inner: Arc<Mutex<Option<E>>>,
+    _e: PhantomData<E2>,
 }
 
-impl<T, Request> Worker<T, Request>
+impl<S, Request, E2> Worker<S, Request, E2>
 where
-    T: Service<Request>,
-    T::Error: Into<crate::BoxError>,
+    S: Service<Request>,
+    S::Error: Into<E2> + Clone,
```

Comment:
I'm really not a fan of adding a `Clone` bound here...in general, I don't think that a majority of the errors being used here will necessarily implement `Clone`, and this feels like it's essentially just punting the complexity of adding the `Arc` to the user?

Reply:
Also, I don't actually see where the `Clone` impl gets used? AFAICT we don't actually need this?

Reply:
it gets used in `process_req`